### PR TITLE
Readme type casting example

### PR DESCRIPTION
**Pull request by borissmidt**
i have added a type casting example to the readme file.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -350,6 +350,14 @@ _Note that prior to release 1.2 of VexCL the `VEX_FUNCTION` macro had different
 interface. That version is considered deprecated but is still available as
 `VEX_FUNCTION_V1`._
 
+Type casting a vector can be done by creating a user defined function. This has the advantage of beeing backend independent.
```
Comment (ddemidov) [inline README.md]: What do you think of replacing the first sentence with `Another example of using a custom function is type-casting a vector`?

File: README.md [hunk 2]
```diff
@@ -350,6 +350,14 @@ _Note that prior to release 1.2 of VexCL the `VEX_FUNCTION` macro had different
...
 `VEX_FUNCTION_V1`._
 
+Type casting a vector can be done by creating a user defined function. This has the advantage of beeing backend independent. 
+~~~{.cpp}
+VEX_FUNCTION(float, make_float, (int, i),
+    return (float)i;
+    );
+b = make_float(a) / 2;
+~~~
+The ~~~{.cpp}vex::convert_XXX~~~ are OpenCL binding and cannot be used with CUDA.
```
Comment (ddemidov) [inline README.md]: Inline code should be enclosed with backticks. Triple tildas [result](https://github.com/borissmidt/vexcl/blob/86fa9d8f34df141b9a62428ecd349d50ce2af945/README.md#user-defined-functions) in striked out text.

Also, I would rephrase the last sentence to read
```
Another option would be to use builtin OpenCL functions [convert_*](https://www.khronos.org/registry/cl/sdk/1.1/docs/man/xhtml/convert_T.html)',
but those are obviously only available for OpenCL backend.
```

**Comment by borissmidt** [comment]
i have included the changes you gave in your comments.

**Comment by ddemidov** [comment]
Thank you! Would it be too much to ask you to squash these two commits together (`git rebase -i HEAD~2`)?

**Comment by borissmidt** [comment]
Of course not, i didn't even know that was possible. I hope it worked.

**Comment by ddemidov** [comment]
It did not. Did you do `git push -f` afterwards?

**Comment by borissmidt** [comment]
i tried it just tells me "everything is up to date".

**Comment by ddemidov** [comment]
When you do `git rebase -i HEAD~3` it should show you text like the following inside a text editor:
```
pick ab927f3 Added type casting to user definded functions
pick 86fa9d8 Added type casting example to user defined functions
pick c7b53be changed the sentenses
```
If you replace `pick` with `squash` in last two lines, it will squash all three commits together (and let you edit the resulting commit message).

**Comment by borissmidt** [comment]
Thank you for your patience. I hope the rebase worked.

**Comment by ddemidov** [comment]
Merged. Thanks again!

**Comment by borissmidt** [comment]
i think it failed... damn it i really suck at git. Hopefully i will be able to use git correctly in the future
